app/controllers/key_pairs_controller.rb

```diff
@@ -0,0 +1,116 @@
+class KeyPairsController < ApplicationController
+  # GET /key_pairs
+  # GET /key_pairs.json
+  def index
+    @key_pairs = KeyPair.all
+
+    respond_to do |format|
+      format.html # index.html.erb
+      format.json { render json: @key_pairs }
+    end
+  end
+
+  # GET /key_pairs/1
+  # GET /key_pairs/1.json
+  def show
+    @key_pair = KeyPair.find(params[:id])
+
+    respond_to do |format|
+      format.html # show.html.erb
+      format.json { render json: @key_pair }
+    end
+  end
+
+  # GET /key_pairs/new
+  # GET /key_pairs/new.json
+  def new
+    @key_pair = KeyPair.new
+
+    respond_to do |format|
+      format.html # new.html.erb
+      format.json { render json: @key_pair }
+    end
+  end
+
+  # GET /key_pairs/1/edit
+  def edit
+    @key_pair = KeyPair.find(params[:id])
+  end
+
+  # Given a File object to upload, save it on the file system with
+  # association to the user_name given.
+  # Admins are able to save keys under different user names then
+  # their own.
+  # Creates the KEY_STORAGE directory if it does not yet exist
+  def upload_key_file(file, user_name)
```

Comment:
Unused method argument - `file`. If it's necessary, use `_` or `_file` as an argument name to indicate that it won't be used.